flink-runtime/src/main/java/org/apache/flink/runtime/blob/FileSystemBlobStore.java

```diff
@@ -81,7 +81,7 @@ public void put(File localFile, JobID jobId, String key) throws Exception {
 	}
 
 	private void put(File fromFile, String toBlobPath) throws Exception {
-		try (OutputStream os = FileSystem.get(new URI(toBlobPath))
+		try (OutputStream os = FileSystem.get(new Path(toBlobPath).toUri())
```

Comment:
Would be good to avoid creating the Path twice and do `Path p = new Path(toBlobPath); p.getFileSystem().create(p, true)`